ontology/obo/tree.go

```diff
@@ -0,0 +1,160 @@
+package obo
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+)
+
+// BuildTree takes a slice of Obo structs and builds a tree, setting
+// the Parents and Children fields to point to related nodes.
+func BuildTree(terms []Obo) map[string]*Obo {
+	termMap := makeTermMap(terms)
+
+	var parentId string
+	var foundInMap bool
+	var parentTerm *Obo
+	var isAItem IsADescription
+
+	// Now we iterate through terms and set up the tree
+	for i := range terms {
+		term := &terms[i] //get a memory copy
```

Comment:
Why do we make a memory copy?  Is it because we may be editing the slice, but want the tree to be separate data that is not linked back to the slice?  If obo files are big, maybe we read it into a slice of pointers to structs, then link up the structs, with no copies.  We then return the tree to the person who asked us to read the file.
I think due to history, we first thought about treating ontologies as a list, rather than a tree.  Now that we have decided to treat them like a tree, it probably makes sense to use a tree data structure.  Right now we have some tension where we are wanting to treat it like a tree, but are holding onto the list structure.